src/main/java/be/seeseemelk/mockbukkit/block/state/BlockStateMock.java

```diff
@@ -420,65 +426,55 @@ else if (MaterialTags.SKULLS.isTagged(block))
-		case DAYLIGHT_DETECTOR:
-			return new DaylightDetectorMock(block);
-		case COMMAND_BLOCK, CHAIN_COMMAND_BLOCK, REPEATING_COMMAND_BLOCK:
-			return new CommandBlockMock(block);
-		case CAMPFIRE, SOUL_CAMPFIRE:
-			return new CampfireMock(block);
-		case BELL:
-			return new BellMock(block);
-		case LECTERN:
-			return new LecternMock(block);
-		case HOPPER:
-			return new HopperMock(block);
-		case BARREL:
-			return new BarrelMock(block);
-		case DISPENSER:
-			return new DispenserMock(block);
-		case DROPPER:
-			return new DropperMock(block);
-		case CHEST, TRAPPED_CHEST:
-			return new ChestMock(block);
-		case ENDER_CHEST:
-			return new EnderChestMock(block);
-		default:
-			return new BlockStateMock(block);
+			case STRUCTURE_BLOCK -> new StructureMock(block);
+			case SMOKER -> new SmokerMock(block);
+			case END_GATEWAY -> new EndGatewayMock(block);
+			case SCULK_CATALYST -> new SculkCatalystMock(block);
+			case SCULK_SHRIEKER -> new SculkShriekerMock(block);
+			case SCULK_SENSOR -> new SculkSensorMock(block);
+			case BEACON -> new BeaconMock(block);
+			case BEEHIVE -> new BeehiveMock(block);
+			case BREWING_STAND -> new BrewingStandMock(block);
+			case BLAST_FURNACE -> new BlastFurnaceMock(block);
+			case COMPARATOR -> new ComparatorMock(block);
+			case CONDUIT -> new ConduitMock(block);
+			case ENCHANTING_TABLE -> new EnchantingTableMock(block);
+			case JIGSAW -> new JigsawMock(block);
+			case JUKEBOX -> new JukeboxMock(block);
+			case SPAWNER -> new CreatureSpawnerMock(block);
+			case DAYLIGHT_DETECTOR -> new DaylightDetectorMock(block);
+			case COMMAND_BLOCK, CHAIN_COMMAND_BLOCK, REPEATING_COMMAND_BLOCK -> new CommandBlockMock(block);
+			case CAMPFIRE, SOUL_CAMPFIRE -> new CampfireMock(block);
+			case BELL -> new BellMock(block);
+			case LECTERN -> new LecternMock(block);
+			case HOPPER -> new HopperMock(block);
+			case BARREL -> new BarrelMock(block);
+			case DISPENSER -> new DispenserMock(block);
+			case DROPPER -> new DropperMock(block);
+			case CHEST, TRAPPED_CHEST -> new ChestMock(block);
+			case ENDER_CHEST -> new EnderChestMock(block);
+			default -> new BlockStateMock(block);
+		};
+	}
+
+	public ItemInteractionResult simulateUseItemOn(BlockStateMock blockState, Location location, ItemStackMock itemStack, PlayerMock playerMock, EquipmentSlot equipmentSlot, BlockFace clickedFace)
+	{
+		if (blockData instanceof BlockDataMock blockDataMock)
+		{
+			return blockDataMock.simulateUseItemOn(blockState, location, itemStack, playerMock, equipmentSlot, clickedFace);
+		}
+		throw new UnimplementedOperationException();
+	}
+
+	public InteractionResult simulateUseWithoutItem(BlockStateMock blockState, Location location, PlayerMock playerMock, BlockFace clickedFace)
+	{
+		if (blockData instanceof BlockDataMock blockDataMock)
+		{
+			return blockDataMock.simulateUseWithoutItem(blockState, location, playerMock, clickedFace);
 		}
+		throw new UnimplementedOperationException();
 	}
```

Comment:
Might be good to make these @ApiStatus.Internal.

Otherwise, if you want these to be part of the api, could you comment them?